packages/integrations/vercel/src/lib/redirects.ts

```diff
@@ -14,6 +14,33 @@ interface VercelRoute {
 	continue?: boolean;
 }
 
+// Copied from astro/packages/astro/src/core/routing/manifest/create.ts
+// Disable eslint as we're not sure how to improve this regex yet
+// eslint-disable-next-line regexp/no-super-linear-backtracking
+const ROUTE_DYNAMIC_SPLIT = /\[(.+?\(.+?\)|.+?)\]/;
+const ROUTE_SPREAD = /^\.{3}.+$/;
+function getParts(part: string, file: string) {
```

Comment:
I've seen this very function used in the Cloudflare adapter too. Maybe we should consider exposing it

Reply:
That's right, it would make sense to expose it in the core package. Maybe also the code for parsing the segments: 
```typescript
const segments = removeLeadingForwardSlash(content)
  .split(nodePath.posix.sep)
  .filter(Boolean)
  .map((s: string) => {
	  return getParts(s, content);
  });
``` 
I've seen it used many times in the core package and the Cloudflare adapter.